lib/renovate.ts

```diff
@@ -1,5 +1,6 @@
 #!/usr/bin/env node
 
+import { getTracer, getTracerProvider } from './instrumentation'; // has to be imported before logger and other libraries which are instrumentalised
```

Comment:
> `// has to be imported before logger and other libraries which are instrumentalised`
I guess you're lucky then that the I comes early in the alphabet.
I'd recommend making the init calls explicitly instead.

Reply:
> A List of all the ModuleSpecifier strings used by the module represented by this record to request the importation of a module. The List is source text occurrence ordered.

https://stackoverflow.com/a/69822046/8111509 and https://tc39.es/ecma262/#table-cyclic-module-fields suggest they are ordered by occurrence as long they are not async modules

The OpenTelemetry examples implement it the same way e.g. https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/examples/memcached.

I have tried the approach you are describing also before my current implementation e.g.      
`lib/renovate.ts`:
```typescript
import { logger } from './logger';
import * as proxy from './proxy';
import * as globalWorker from './workers/global';
import { getTracer, getTracerProvider, init as initOpenTelemetry } from './instrumentation';

initOpenTelemetry()
```
and    
`lib/instrumentation/index.ts`:
```typescript 
let traceProvider: NodeTracerProvider;

//   init();

export function init(): NodeTracerProvider {
 //  ....
}
```
In that case the logging instrumentation is not applied.